### DISCUSSION TITLE
Input values in single step debugging

### DISCUSSION THREAD

**Discussion by gbizzotto** [hneemann/Digital#1049] [Q&A]
In single step debugging, when i change an input value (that is, more than one bit), it fully propagates, skipping the step by step feature. I need to change inputs at different steps (for instance, one input before the first step, another input before the second step, another one before the fourth, etc), and those can't interact properly otherwise.

Did i miss a config that enables step by step for input value changes?

**Edit**: As a workaround, one could, for each such input, add a second 1-bit input to switch it on and off, so the many-bits value can propagate properly, like so:
![Screenshot from 2022-11-04 20-43-21](https://user-images.githubusercontent.com/5064376/200090016-6716f193-5c55-4c08-ab84-9705483905c1.png)

Of course, this is additional work before and after debug. Furthermore, it adds one step before the initial propagation of the value.

**Comment by hneemann** [comment]
You can change a value, perform a microstep, and then change the next value. And the same goes when changing one bit of a value.

**Comment by gbizzotto** [comment]
As soon as i change the value, it's already propagated all the way.
https://user-images.githubusercontent.com/5064376/200818766-10978f5f-6156-473a-9013-c2a83ac47ca3.mp4

**Comment by hneemann** [comment]
You need to start the simulation in "single gate mode" if you want to follow the propagation of a signal.

**Comment by gbizzotto** [comment]
Isn't that what i did? "Single gate step mode"?

**Comment by hneemann** [comment]
@gbizzotto 
I couldn't view your video, but I tried it myself and there is an issue. I have to look at it.

**Comment by hneemann** [comment]
see #1053

**Comment by hneemann** [comment]
@gbizzotto 
Please try the pre release [v0.29-56-g500277d ](https://infdigital.dhbw-mosbach.de/)

**Comment by gbizzotto** [comment]
Fixed. Thanks!

**Reply by hneemann** [reply]
Thank's for pointing this out! :smile:
